### PR TITLE
Configurable I2C clock stretching

### PR DESCRIPTION
**Pull request by UncleRus**
Fix for #471

### CODE REVIEW

File: extras/i2c/i2c.h [hunk 1]
```diff
@@ -41,9 +41,11 @@ extern "C" {
  * Define i2c bus max number
  */
 #ifndef MAX_I2C_BUS
-    #define MAX_I2C_BUS 2
+    #define I2C_MAX_BUS 2
```
Comment (ourairquality) [inline extras/i2c/i2c.h]: Would want to change the `#ifndef MAX_I2C_BUS` too here.

Also this patch will need to be edited to remove the lwip changes. If working with modules is a challenge then just save the diffs to a file and remove all the changes from this branch then re-apply and make a forced push. Then just don't commit the lwip directory.

Comment (UncleRus) [reply extras/i2c/i2c.h]: Thank you for your review!

**Comment by UncleRus** [comment]
Going to merge this one if no objections.
@gschorcht any comments?

**Comment by gschorcht** [comment]
@UncleRus 
Many thanks for your changes and all the work you have done in conjunction with them. Perhaps it would be better to leave function ```i2c_init``` as it is and to add a new function ```i2c_set_clock_stretching (bus, clock_stretch_counter)```?

```
i2c_init(I2C_BUS, I2C_SCL_PIN, I2C_SDA_PIN, I2C_FREQ_100K);
i2c_set_clock_stretching (I2C_BUS, MY_CLK_STRETCH);
```

Advantage would be that existing drivers and examples which are using i2c wouldn't have to be changed. Furthermore, clock stretching counter could be changed during runtime if neccessary.

**Pull request merged**